src/macrofier/types.mjs

```diff
@@ -137,6 +139,39 @@ const insertEnumMacros = (content, schema, module, name) => {
   return template.join('\n')
 }
 
+const insertObjectAdditionalPropertiesMacros = (content, schema, module, title, options) => {
+  options = options ? JSON.parse(JSON.stringify(options)) : {}
+
+  const options2 = JSON.parse(JSON.stringify(options))
+  options2.parent = title
+  options2.level = options.level + 1
+  options2.name = ''
+
+  const shape = getSchemaShape(schema.additionalProperties, module, options2)
+  let type = getSchemaType(schema.additionalProperties, module, options2).trimEnd()
+  const propertyNames = localizeDependencies(schema, module).propertyNames
+
+  let jsonType = getJsonType(schema.additionalProperties, module)
+  if (!isPrimitiveType(jsonType)) {
+    jsonType = 'string'
+  }
+
+  const additionalType = getPrimitiveType(jsonType, 'additional-types')
+  let key = (propertyNames && propertyNames.title) ? propertyNames.title : getTemplate(path.join(options.templateDir, 'additionalPropertiesKey')).trimEnd()
+
+  content = content
+    .replace(/\$\{shape\}/g, shape)
```

Comment:
Did you confirm that none of the JS templates were using `${property}` or `${Property}` here?

I forget why I had those two lines in there, but there must have been some reason.

Reply:
I have not seen any use case from javascript. javascript generation not showed any issue